src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php

```diff
@@ -127,7 +128,7 @@ public function collect(Request $request, Response $response, \Exception $except
 
             $logoutUrl = null;
             try {
-                if (null !== $this->logoutUrlGenerator) {
+                if (null !== $this->logoutUrlGenerator && !$token instanceof AnonymousToken) {
```

Comment:
Shouldn't this use the authentication trust resolver (at least in Symfony 4, until this is removed in Sf 5)?

Reply:
The reasoning is that new code doesn't have to support legacy behaviors.
Especially here, since it's a DataCollector: we don't really care.